src/components/Semester.vue

```diff
@@ -194,10 +212,24 @@ export default {
     isSameYear: function () {
       return Math.floor((this.index - 1) / 3) === Math.floor((this.currentSemester - 1) / 3);
     },
+    noLongerOffered: function () {
+      if (this.itemAdding.is_historical) {
+        const lastSemester = this.itemAdding.source_semester.split('-');
+        const sourceSemester = ['fall', 'IAP', 'spring'].indexOf(lastSemester[0]);
+        // which class year the last year offered corresponds to; +1 if fall because fall semester year is off by 1
+        const sourceYear = parseInt(lastSemester[1]) - this.baseYear + (sourceSemester === 0 ? 1 : 0);
+        const lastSemesterNumber = sourceYear * 3 + sourceSemester + 1;
+        if (this.index > lastSemesterNumber) {
+          return true;
+        }
+      }
+      return false;
+    },
```

Comment:
I feel like this `noLongerOffered` function could be used up above where the exact same code is copied for adding the warning (with some small modification to say what year, etc.)